src/core/ddsc/src/dds__sysdef_validation.h

```diff
@@ -0,0 +1,29 @@
+// Copyright(c) 2024 ZettaScale Technology and others
+//
+// This program and the accompanying materials are made available under the
+// terms of the Eclipse Public License v. 2.0 which is available at
+// http://www.eclipse.org/legal/epl-2.0, or the Eclipse Distribution License
+// v. 1.0 which is available at
+// http://www.eclipse.org/org/documents/edl-v10.php.
+//
+// SPDX-License-Identifier: EPL-2.0 OR BSD-3-Clause
+#ifndef DDS__SYSDEF_VALIDATION_H
+#define DDS__SYSDEF_VALIDATION_H
+
+#if defined (__cplusplus)
+extern "C" {
+#endif
+
+#include "dds/dds.h"
+#include "dds__sysdef_model.h"
+
+#define SYSDEF_QOS_USERDATA_MAXLEN ((size_t) 255)
```

Comment:
I don't see why we would hard-code a maximum length in the QoS provider for user data and partition names. (There is a hard limit in the discovery message format, but that is 64kB-1.) If we drop it here, we also no longer need the check elsewhere.